deployment/ccip/changeset/solana/cs_ping_pong.go

```diff
@@ -0,0 +1,254 @@
+
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to build instruction: %w", err)
+	}
+
+	if err := chain.Confirm([]solana.Instruction{initTx}); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to confirm initializeRMNRemote: %w", err)
+	}
+
+	e.Logger.Infow("Initialized ping pong demo", "chain", chain.String())
+
+	return deployment.ChangesetOutput{
+		AddressBook: newAddresses,
+	}, nil
+}
+
+type StartPingPongConfig struct {
+	ChainSelector uint64
+}
+
+func (c StartPingPongConfig) Validate(e deployment.Environment) error {
+	return nil
+}
+func StartPingPongContractChangeset(
+	e deployment.Environment,
+	c StartPingPongConfig,
+) (deployment.ChangesetOutput, error) {
+	if err := c.Validate(e); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid StartPingPongConfig: %w", err)
+	}
+
+	s, err := ccipChangeset.LoadOnchainStateSolana(e)
+	if err != nil {
+		e.Logger.Errorw("Failed to load existing onchain state", "err", err)
+		return deployment.ChangesetOutput{}, err
+	}
+
+	chain := e.SolChains[c.ChainSelector]
+	chainState := s.SolChains[c.ChainSelector]
+
+	ppConfigPDA, _, err := solanaStateUtils.FindPingPongDemoConfigPDA(chainState.PingPong)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	ppSendSignerPDA, _, err := solanaStateUtils.FindPingPongCCIPSendSignerPDA(chainState.PingPong)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	destChainStatePDA, err := solanaStateUtils.FindDestChainStatePDA(c.ChainSelector, chainState.Router)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	routerNoncePDA, err := solanaStateUtils.FindNoncePDA(c.ChainSelector, solana.PublicKey(*chain.DeployerKey), chainState.Router)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	feeBillingSignerPDA, _, err := solanaStateUtils.FindFeeBillingSignerPDA(chainState.Router)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	fqBillingTokenConfigPDA, _, err := solanaStateUtils.FindFqBillingTokenConfigPDA(nil, chainState.FeeQuoter)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	rmnRemoteConfigPDA, _, err := solanaStateUtils.FindRMNRemoteConfigPDA(chainState.RMNRemote)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	rmnRemoteCursesPDA, _, err := solanaStateUtils.FindRMNRemoteCursesPDA(chainState.RMNRemote)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to find ping pong config pda: %w", err)
+	}
+
+	initTx, err := solPingPong.NewStartPingPongInstruction(
```

Comment:
For reference, here's an integration test using this instruction, and you can check there how each field is calculated: https://github.com/smartcontractkit/chainlink-ccip/blob/d2fd3633f4854492d233a0b545dd17895cfcdc4e/chains/solana/contracts/tests/examples/ping_pong_test.go#L448-L471